cwm_worker_tests/upload_download_test.py

```diff
@@ -95,78 +109,97 @@ def run(self):
                 print(f'Bucket does not exist! [{self.bucket}]')
                 return
 
-            iterations = []
             for iteration in range(self.download_iterations):
                 iteration_no = iteration + 1
                 iteration_start_time = datetime.datetime.now()
                 print(f'{context} | Iteration # {iteration_no} | Started')
-                objects = client.list_objects(self.bucket, recursive=True)
-                num_downloaded_files = 0
-                for obj in objects:
-                    object_name = obj.object_name
-                    object_filepath = f'/tmp/{object_name}.{self.threadid}' 
-                    client.fget_object(self.bucket, object_name, object_filepath)
-                    num_downloaded_files += 1
-                    os.remove(object_filepath)
-                # print(f'{context} | Downloaded: {num_downloaded_files} files', end='', flush=True)
+                for i in range(self.num_files):
+                    download_start_time = datetime.datetime.now()
+                    error = None
+                    try:
+                        index = i+1
+                        object_name = get_filename(self.file_size, index)
+                        object_filepath = f'/tmp/{object_name}.{self.threadid}'
+                        client.fget_object(self.bucket, object_name, object_filepath)
+                        object_size_bytes = os.path.getsize(object_filepath)
+                        os.remove(object_filepath)
+                    except S3Error as e:
+                        error = str(e)
+                    except Exception as e:
+                        error = str(e)
```

Comment:
except Exception catches all errors, no need to have an additional except S3Error